tabletools/table_paste_col.py

```diff
@@ -1,45 +1,47 @@
 #!/usr/bin/env python
-
-"""
-Add a column containing a constant value to a table. Header assumed.
-"""
-
 import sys
-import argparse 
+import argparse
 
 def get_input_file_object(filename):
     if filename == "-":
         return sys.stdin
     return open(filename, "r")
 
-parser = argparse.ArgumentParser(description=__doc__)
-parser.add_argument("-t", "--table", default='-', help="input csv file; STDIN if - (default: %(default)s")
-parser.add_argument("-c", "--col-name", required=True, help="name for new column")
-parser.add_argument("-v", "--col-value", required=True, help="constant value for new column")
-parser.add_argument("-s", "--separator", default='\t', help="column separator character, default: <TAB>")
-parser.add_argument("--at-end", action="store_true", help="inserts new column as the right most one.")
-args = parser.parse_args()
-
+def parse_args(args):
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-t", "--table", default='-', help="input csv file; STDIN if - (default: %(default)s)")
+    parser.add_argument("-c", "--col-name", required=True, help="name for new column")
+    parser.add_argument("-v", "--col-value", required=True, help="constant value for new column")
+    parser.add_argument("-s", "--separator", default='\t', help="column separator character, default: <TAB>")
+    parser.add_argument("--at-end", action="store_true", help="inserts new column as the right most one.")
+    return parser.parse_args(args)
 
-table = get_input_file_object(args.table)
-    
-def add_col(sep, colname, colvalue, at_end):
+def add_col(table, sep, colname, colvalue, at_end, out=sys.stdout):
     i = 0
     for line in table:
+        line = line.rstrip('\n')
         if i==0:
-            line = line.rstrip('\n')
             if at_end:
-                print(line + sep + colname)
+                print(line + sep + colname, file=out)
             else:
-                print(colname + sep + line)
+                print(colname + sep + line, file=out)
         else:
-            line = line.rstrip('\n')
             if at_end:
-                print(line + sep + colvalue)
+                print(line + sep + colvalue, file=out)
             else:
-                print(colvalue + sep + line)
+                print(colvalue + sep + line, file=out)
         i += 1
-
+    return 
+
+def main():
+    """
+    Add a column containing a constant value to a table. Header assumed.
+    """
```

Comment:
This needs to be at the top of the file. When it's at the top it becomes the docstring for the package and is assigned to the special variable `__doc__`. You use this variable as documentation on ln 11. As is, this variable will be empty and the help message won't print correctly when `--help` is called.